www/front_src/src/Navigation/Sidebar/helpers/getUrlFromEntry.ts

```diff
@@ -0,0 +1,50 @@
+import { equals, isNil } from 'ramda';
+
+import { Page } from '../../models';
+
+export const getUrlFromEntry = ({
+  page,
+  options,
+  is_react,
+  url,
+}: Page): string | null | undefined => {
+  const currentPage = isNil(page) ? '' : page;
+  const currentOptions = isNil(options) ? '' : options;
+
+  const urlOptions = `${currentPage}${currentOptions}`;
+  const currentUrl = is_react ? url : `/main.php?p=${urlOptions}`;
+
+  return currentUrl;
+};
+
+const isArrayItem = (item: unknown): boolean => {
+  if (isNil(item) || !Array.isArray(item) || equals(item?.length, 0)) {
+    return false;
+  }
+
+  return true;
+};
+
+export const searchUrlFromEntry = (item: Page): string | null | undefined => {
+  const childPage = item?.children;
+  const groupPage = item?.groups;
+
+  if (isArrayItem(childPage) && Array.isArray(childPage)) {
```

Comment:
I think the `Array.isArray(childPage)` is overkill here because `isArrayItem` will return `true` if its an array